infra/base-images/base-builder/compile_libfuzzer

```diff
@@ -16,16 +16,22 @@
 ################################################################################
 
 echo -n "Compiling libFuzzer to $LIB_FUZZING_ENGINE... "
-mkdir -p $WORK/libfuzzer
-pushd $WORK/libfuzzer > /dev/null
+export LIB_FUZZING_ENGINE="-fsanitize=fuzzer"
+if  [ "$SANITIZER" = "address" ] || [ "$SANITIZER" = "undefined" ]
+then
+  cp /usr/local/lib/clang/12.0.0/lib/linux/libclang_rt.fuzzer-$ARCHITECTURE.a \
+    $LIB_FUZZING_ENGINE_DEPRECATED
+else
+  mkdir -p $WORK/libfuzzer
+  pushd $WORK/libfuzzer > /dev/null
 
-# Use -fPIC to allow preloading (LD_PRELOAD).
-$CXX $CXXFLAGS -std=c++11 -O2 -fPIC $SANITIZER_FLAGS -fno-sanitize=vptr \
+  # Use -fPIC to allow preloading (LD_PRELOAD).
```

Comment:
I wonder if my change will break people who use LD_PRELOAD and libFuzzingEngine.a

Reply:
Can we verify this? If LD_PRELOAD currently already works for people who use `-fsanitize=fuzzer`, then there shouldn't be any reason this would break?

Reply:
Yes good point.